### PR TITLE
Simple Control Add-in Sample (with X-mas theme)

### PR DESCRIPTION
**Pull request by niku-live**
Since now it is possible to create control add-ins with AL language, there should be a sample that would demonstrate this. I created a sample with extends Customer List page with additional part in FactBoxes where pretty X-mas greeting (with snow) is displayed. I think it's good sample in time for Christmas.

### CODE REVIEW

File: samples/XMasControlAddIn/AddIn/AddInInterface.al [hunk 1]
```diff
@@ -0,0 +1,18 @@
+controladdin NavSnowControlAddIn
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/AddIn/AddInInterface.al]: I think AddInInterface.al should be named like the class itself, NavSnowControlAddIn.

File: samples/XMasControlAddIn/Pages/XmasGreetingPart.al [hunk 2]
```diff
@@ -0,0 +1,33 @@
+page 50101 "X-mas Greeting Part"
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Pages/XmasGreetingPart.al]: I would prefer to call it XmasGreetingPart and use the nicer name in the caption. This fits better to the new language model

````VisualBasic
page 50101 XmasGreetingPart
````

File: samples/XMasControlAddIn/Pages/XmasGreetingPart.al [hunk 3]
```diff
@@ -0,0 +1,33 @@
+page 50101 "X-mas Greeting Part"
+{
+    PageType = CardPart;
+    SourceTable = Customer;
+
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Pages/XmasGreetingPart.al]:
````VisualBasic
Caption = "X-mas Greeting Part";
````

**Comment by niku-live** [comment]
Thanks for good suggestions. I added changes, also added translations to en-US and my native language lt-LT.

File: samples/XMasControlAddIn/AddIn/AddInInterface.al [hunk 4]
```diff
@@ -0,0 +1,18 @@
+controladdin NavSnowControlAddIn
+{
+    RequestedHeight = 300;    
+    RequestedWidth = 300;
+    VerticalStretch = true;
+    VerticalShrink = true;
+    HorizontalStretch = true;
+    HorizontalShrink = true;
+    Scripts =         
+        'AddIn/jquery.min.js',
```
Comment (dennisfrostlander) [inline samples/XMasControlAddIn/AddIn/AddInInterface.al]: We could use a reference directly to the CDN, for example:
http://ajax.aspnetcdn.com/ajax/jQuery/jquery-3.2.1.min.js

Then we can demonstrate another feature and don't need to keep a copy of jquery in our samples folder

Comment (niku-live) [reply samples/XMasControlAddIn/AddIn/AddInInterface.al]: To tell you the truth I didn't knew that CDN links was supported. That's great new knowledge to me, thank. I replaced jQuery with CDN link

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="en-US" original="XMasControlAddIn">
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf]: The target language is missing.

````XML
<file datatype="xml" source-language="en-US" original="XMasControlAddIn" target-language="en">
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="en-US" original="XMasControlAddIn">
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Christmas Greeting</source>
+          <note from="Developer" annotates="general" priority="2" />
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf]: The translation is missing here.

````XML
<target state="translated">Christmas Greeting</target><note from="Developer" annotates="general" priority="2"/>
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Christmas Greeting</source>
+          <note from="Developer" annotates="general" priority="2" />
+          <note from="Xliff Generator" annotates="general" priority="3">PageExtension - Control</note>
+        </trans-unit>
+        <trans-unit id="PageExtension 1255613137 - NamedType 1965138970" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Happy Christmas, %1</source>
+          <note from="Developer" annotates="general" priority="2" />
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.en-US.xlf]: The translation is missing here.

````XML
<target state="translated">Happy Christmas, %1</target><note from="Developer" annotates="general" priority="2"/>
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf [hunk 8]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="lt-LT" original="XMasControlAddIn">
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf]: The target language is missing.

````XML
<file datatype="xml" source-language="en-US" original="XMasControlAddIn" target-language="lt">
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf [hunk 9]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="lt-LT" original="XMasControlAddIn">
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Kalėdų sveikinimas</source>
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf]: Source is not english.

````XML
<source>Christmas Greeting</source>
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf [hunk 10]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="lt-LT" original="XMasControlAddIn">
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Kalėdų sveikinimas</source>
+          <note from="Developer" annotates="general" priority="2" />
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf]: The translation is missing here.

````XML
<target state="translated">Kalėdų sveikinimas</target><note from="Developer" annotates="general" priority="2"/>
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Kalėdų sveikinimas</source>
+          <note from="Developer" annotates="general" priority="2" />
+          <note from="Xliff Generator" annotates="general" priority="3">PageExtension - Control</note>
+        </trans-unit>
+        <trans-unit id="PageExtension 1255613137 - NamedType 1965138970" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Laimingų šv. Kalėdų, %1</source>
+          <note from="Developer" annotates="general" priority="2" />
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf]: The translation is missing here.

````XML
<target state="translated">Laimingų šv. Kalėdų, %1</target><note from="Developer" annotates="general" priority="2"/>
````

File: samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  <file datatype="xml" source-language="lt-LT" original="XMasControlAddIn">
+    <body>
+      <group id="body">
+        <trans-unit id="PageExtension 1255613137 - Control 657769462 - Property 2879900210" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Kalėdų sveikinimas</source>
+          <note from="Developer" annotates="general" priority="2" />
+          <note from="Xliff Generator" annotates="general" priority="3">PageExtension - Control</note>
+        </trans-unit>
+        <trans-unit id="PageExtension 1255613137 - NamedType 1965138970" maxWidth="999" size-unit="char" translate="yes" xml:space="preserve">
+          <source>Laimingų šv. Kalėdų, %1</source>
```
Comment (Gallimathias) [inline samples/XMasControlAddIn/Translations/XMasControlAddIn.lt-LT.xlf]: Source is not english.

````XML
<source>Happy Christmas, %1</source>
````

**Comment by niku-live** [comment]
Well, I am still learning about this XLIFF file :D. I hope now it is correct.

**Comment by Gallimathias** [comment]
No problem. In addition, the community is there for mutual learning :)

**Comment by StanislawStempin** [comment]
Our apologies for the delay. Since this pull request contains 3rd party code with unclear licensing rights we have to unfortunately reject this contribution.

**Pull request closed without merging**